packages/@react-stately/color/test/useHexColorFieldState.test.js

```diff
@@ -0,0 +1,218 @@
+    expect(result.current.inputValue).toBe('#AABBC8');
+    expect(result.current.validationState).toBe(null);
+  });
+
+  it('should not decrement beyond the minimum value', function () {
+    let props = {defaultValue: '#abc', minValue: '#abc'};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.decrement());
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#AABBCC');
+  });
+
+  it('should decrement to min value', function () {
+    let props = {defaultValue: new Color('#aabbcc'), minValue: '#666'};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.decrementToMin());
+    expect(result.current.colorValue.value).toEqual({
+      red: 102,
+      green: 102,
+      blue: 102,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#666666');
+  });
+
+  it('should revert to last valid value', function () {
+    let props = {defaultValue: '#abc', minValue: '#abc'};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.decrement());
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#AABBCC');
+  });
+
+  it('should update colorValue (uncontrolled)', function () {
+    const onChangeSpy = jest.fn();
+    let props = {defaultValue: '#abc', minValue: '#abc', onChange: onChangeSpy};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#AABBCC');
+
+    const newColor = new Color('#cba');
+    act(() => result.current.setColorValue(newColor));
+    expect(onChangeSpy).toHaveBeenCalledWith(newColor);
+    expect(result.current.colorValue.value).toEqual({
+      red: 204,
+      green: 187,
+      blue: 170,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#CCBBAA');
+  });
+
+  it('should not update colorValue (controlled)', function () {
+    const onChangeSpy = jest.fn();
+    let props = {value: '#abc', onChange: onChangeSpy};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#AABBCC');
+
+    const newColor = new Color('#cba');
+    act(() => result.current.setColorValue(newColor));
+    expect(onChangeSpy).toHaveBeenCalledWith(newColor);
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+    // expect(result.current.inputValue).toBe('#CCBBAA');
```

Comment:
I hope I'm understanding the concept of controlled states correctly.

Here, `colorValue` is controlled so it doesn't change the `colorValue` state even after `setColorValue` is called.
Instead, it will rely on the component to be re-rendered for it to reflect the new color value.

I'm not sure how the other states like `inputValue` should respond. It currently still updates to the new color we provide in `setColorValue`. Maybe I should put it in a `useEffect` so `inputValue` changes only when `colorValue` changes.

Reply:
Resolved. Will revisit this question later if it still makes sense.